### PR TITLE
feat(eslint-plugin): add `no-typeof` rule

### LINKED ISSUE #3185: 🤖 - replace `typeof etc === 'string'` to `tuiIsString(etc)` by ESlint
### Description

image

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #3185

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/icons/scripts/process-icons.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+// eslint-disable-next-line @taiga-ui/no-deep-imports
+import {tuiIsString} from '@taiga-ui/cdk/utils';
```
Comment (nsbarsukov) [inline projects/icons/scripts/process-icons.ts:2]: package `@taiga-ui/icons` has no deps.
I think we should exclude this rule for `projects/icons` and this line

File: projects/testing/cypress/snapshot/command.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 /// <reference types="cypress" />
-
+// eslint-disable-next-line @taiga-ui/no-deep-imports
+import {tuiIsObject, tuiIsString} from '@taiga-ui/cdk/utils/miscellaneous';
```
Comment (nsbarsukov) [inline projects/testing/cypress/snapshot/command.ts:3]: Why deep import ?

I think Ivy can tree-shake it.

Comment (splincode) [reply projects/testing/cypress/snapshot/command.ts:3]: image

File: projects/icons/scripts/process-icons.ts (hunk 1 above)
Comment (splincode) [reply projects/icons/scripts/process-icons.ts:2]: image

**Review by nsbarsukov — APPROVED** [review]

**Comment by splincode** [comment]
@vladimirpotekhin updated

**Review by waterplea — APPROVED** [review]

**Pull request merged**